packages/api/src/routers/api/clickhouseProxy.ts

```diff
@@ -162,6 +162,12 @@ const proxyMiddleware: RequestHandler =
           proxyRes.headers['access-control-allow-origin'] = _req.headers.origin;
           proxyRes.headers['access-control-allow-credentials'] = 'true';
         }
+
+        // Add a custom header to indicate that the response is a mixed response when applicable
+        // since the Clickhouse Web SDK allows accessing headers but not status codes.
+        if (proxyRes.statusCode === 207) {
+          proxyRes.headers['X-ClickHouse-Mixed-Response'] = 'true';
```

Comment:
nit: I might consider naming this `X-ClickStack-....`. Using the name 'ClickHouse' might make it seem like the header was added by the CH server when it's really an artifact of our application. I can just see some troubleshooting getting complex if we can't figure out where it comes from.